### PR TITLE
finished 164

### PR DESCRIPTION
**Pull request by JoannaCode**
@FreeTymeKiyan

### CODE REVIEW

File: Hard/MaximumGap_shuna.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        int length = nums.length;
+        int maxGap = 0;
+        for(int i : nums){
+           max = Math.max(max, i);
+           min = Math.min(min, i);
+        }
+        if(min == max){
+            return 0;
+        }
+        int bicketSize = (max - min) / (length - 1);
```
Comment (FreeTymeKiyan) [inline Hard/MaximumGap_shuna.java]: bicket? bucket?

File: Hard/MaximumGap_shuna.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        if(bicketSize == 0){
+            bicketSize = 1;
+        }
+        int bicketCount = (max - min) / bicketSize + 1;
```
Comment (FreeTymeKiyan) [inline Hard/MaximumGap_shuna.java]: same here

File: Hard/MaximumGap_shuna.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        int[] left = new int[bicketCount];
+        int[] right = new int[bicketCount];
+        Arrays.fill(left, -1);
+        Arrays.fill(right, -1);
+        for(int i = 0; i < length ; i ++){
```
Comment (FreeTymeKiyan) [inline Hard/MaximumGap_shuna.java]: space between i and ++

File: Hard/MaximumGap_shuna.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            int loc = (nums[i] - min) / bicketSize;
+            if(left[loc] == -1){
+               left[loc] = nums[i];
```
Comment (FreeTymeKiyan) [inline Hard/MaximumGap_shuna.java]: only 3 spaces here

**Comment by FreeTymeKiyan** [comment]
You can make use of the formatting function of whatever the IDE you are using right now. This way you won't be troubled by any format problems anymore.

**Comment by JoannaCode** [comment]
@FreeTymeKiyan fixed~

**Comment by FreeTymeKiyan** [comment]
:+1:

**Pull request merged**